phylogeny/tests/test_prototypeSelection.py

```diff
@@ -0,0 +1,143 @@
+        self.assertRaisesRegex(
+            DissimilarityMatrixError,
+            'IDs must be unique. Found the following duplicate IDs',
+            distanceSum,
+            ['A', 'B', 'C', 'D', 'B'],
+            self.dm20)
+
+        # test that list of IDs holds at least 1 element
+        self.assertRaisesRegex(
+            DissimilarityMatrixError,
+            'Data must be at least 1x1 in size',
+            distanceSum,
+            [],
+            self.dm20)
+
+        # test for result correctness
+        self.assertAlmostEqual(2454.1437464961, distanceSum(self.dm100.ids,
+                                                            self.dm100))
+        self.assertAlmostEqual(32.9720926186, distanceSum(
+            ['550.L1S173.s.1.sequence', '550.L1S141.s.1.sequence',
+             '550.L1S18.s.1.sequence', '550.L1S156.s.1.sequence',
+             '550.L1S110.s.1.sequence', '550.L1S143.s.1.sequence',
+             '550.L1S134.s.1.sequence', '550.L1S103.s.1.sequence',
+             '550.L1S185.s.1.sequence', '550.L1S114.s.1.sequence',
+             '550.L1S138.s.1.sequence', '550.L1S137.s.1.sequence'],
+            self.dm100))
+
+        self.assertAlmostEqual(81.6313, distanceSum(self.dm20.ids,
+                                                    self.dm20))
+        self.assertAlmostEqual(13.3887, distanceSum(
+            ['A', 'C', 'F', 'G', 'M', 'N', 'P', 'T'],
+            self.dm20))
+
+    def test_exhaustive(self):
+        # check if execution is rejected if number of combination is too high
+        self.assertRaisesRegex(
+            Exception,
+            'Cowardly refuse to test ',
+            prototypeSelection_exhaustive,
+            self.dm20,
+            5,
+            maxCombinationsToTest=1000)
+
+        res = prototypeSelection_exhaustive(self.dm20, 3)
+        self.assertCountEqual(('A', 'P', 'Q'), res)
+        self.assertAlmostEqual(1.841, distanceSum(res, self.dm20))
+
+        res = prototypeSelection_exhaustive(self.dm20, 4)
+        self.assertCountEqual(('A', 'J', 'P', 'T'), res)
+        self.assertAlmostEqual(3.4347, distanceSum(res, self.dm20))
+
+        res = prototypeSelection_exhaustive(self.dm20, 5)
+        self.assertCountEqual(('A', 'C', 'O', 'P', 'T'), res)
+        self.assertAlmostEqual(5.4494, distanceSum(res, self.dm20))
+
+        res = prototypeSelection_exhaustive(self.dm20, 18)
+        self.assertCountEqual(
+            ('A', 'B', 'C', 'D', 'E', 'F', 'G', 'I', 'J', 'K', 'L', 'M', 'N',
+             'O', 'P', 'Q', 'R', 'T'),
+            res)
+        self.assertAlmostEqual(66.94, distanceSum(res, self.dm20))
+
+        res = prototypeSelection_exhaustive(self.dm20, 19)
+        self.assertCountEqual(
+            ('A', 'B', 'C', 'D', 'E', 'F', 'G', 'I', 'J', 'K', 'L', 'M', 'N',
+             'O', 'P', 'Q', 'R', 'S', 'T'),
+            res)
+        self.assertAlmostEqual(74.1234, distanceSum(res, self.dm20))
+
+    def test_wellformedDistanceMatrix(self):
```

Comment:
It would be more appropriate to push theses tests into scikit-bio as a problem there impacts many more users than a problem here; catching it at a lower level would be better. The test suite for this object is already [comprehensive](https://github.com/biocore/scikit-bio/blob/master/skbio/stats/distance/tests/test_base.py), although any corner cases not caught would be welcomes. 

Scikit-bio takes an aggressive approach to regression testing, so it would be a catastrophic failure on the review and build process for a release to come out with a bug along these. If one were to happen, it would almost certainly trigger an immediate bugfix release. 

Unit tests are pragmatic. There are an infinite number of tests that can be performed on even the simplest of programs but that does not mean they are all equally worth the developer effort to put in place and maintain.